core/src/map.cpp

```diff
@@ -155,8 +151,10 @@ SceneID Map::Impl::loadScene(SceneOptions&& _sceneOptions) {
 
 SceneID Map::Impl::loadSceneAsync(SceneOptions&& _sceneOptions) {
 
-    // Avoid to keep loading old scene and tiles
-    auto oldScene = std::move(scene);
+    // Move the previous scene into a shared_ptr so that it can be captured in a std::function
+    // (unique_ptr can't be captured because std::function is copyable).
+    std::shared_ptr<Scene> oldScene = std::move(scene);
```

Comment:
Wow that's interesting. I am not sure if this is a recommended practice but also don't know what's another solution for this. Initial usage of shared ptr for the scene itself was to get around this.. but it came with  it's own tentacles.

Reply:
Yeah I thought about this bit a lot. `unique_ptr` is a better fit for ownership of the scene, it just makes async disposal messier. What would be ideal here is a non-copyable version of `std::function` (there are implementations of this in other projects) but using a `shared_ptr` here seemed like a decent balance between elegance and pragmatism.